<a href="https://colab.research.google.com/github/mendozagenzo3-cmyk/Prog_Civil_2026_1/blob/main/Copia_de_2_3_4_Actividad_Pluma_de_Agua_Algoritmo_GitHub.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# NOMBRES COMPLETOS: Genzo Daniel Mendoza Martínez
# Código: 1103501854

# Actividad de Programación con Python  
## **Historia, ciencia y programación: la “pluma de agua”**

> **Relato de partida**  
> **¿Por qué los costeños le decimos “pluma” a la llave del agua?**  
>  
> En el Caribe colombiano, abrir la “pluma” no es solo una acción cotidiana: es una expresión cargada de historia, técnica y cultura. Aunque algunos se sorprendan o incluso se burlen del término, lo cierto es que decir “pluma” en vez de “llave” no es un error, sino una herencia viva que conecta nuestro lenguaje con siglos de evolución hidráulica.  
>  
> Mucho antes de que existieran los grifos modernos, el agua se distribuía mediante sistemas regulados con precisión. En España, por ejemplo, se usaba la **“pluma de agua”** como unidad de medida para calcular el caudal que se entregaba a cada hogar. Aproximadamente, una pluma equivalía a **0.025 litros por segundo**. En algunos lugares, ese flujo se regulaba mediante orificios calibrados y dispositivos llamados **plumeros**, que permitían controlar la salida del agua.  
>  
> Durante la época colonial, muchas de estas prácticas técnicas cruzaron el Atlántico y llegaron a ciudades como Cartagena, Barranquilla y Santa Marta. Con el tiempo, el término “pluma” dejó de ser solamente técnico y pasó al habla cotidiana como sinónimo de la llave del agua.  
>  
> Así, una palabra del uso diario conserva una memoria histórica, tecnológica y cultural.  
>  
> ---
>  
> **Idea central para esta actividad:**  
> A partir de este relato, construiremos un problema de programación donde una magnitud histórica y física, el **caudal de una pluma de agua**, se represente mediante variables, operaciones y estructuras básicas de Python.

## 1. Propósito de la actividad

Al finalizar esta actividad, el estudiante debe ser capaz de:

- interpretar un contexto histórico y científico como un problema computacional;
- identificar entradas, procesos y salidas;
- diseñar un algoritmo antes de programar;
- implementar el programa en Python;
- publicar el trabajo en GitHub;
- ampliar la documentación del código;
- clonar el repositorio al disco local usando Git.

> **Importante:**  
> En este notebook **no se entrega la solución completa**.  
> Aquí se presenta el **planteamiento**, la **estructura del problema** y el **algoritmo guía** para que cada estudiante construya el código.

## 2. Situación problema

Un hogar recibe una cierta cantidad de **plumas de agua**.  
Se sabe que:

- **1 pluma = 0.025 litros por segundo**

Con esta información, el estudiante debe construir un programa que permita calcular:

1. el caudal total en litros por segundo;
2. el consumo diario en litros;
3. el consumo mensual en litros (suponiendo 30 días);
4. una clasificación del consumo:
   - **Bajo**: menos de 1000 litros por día
   - **Medio**: entre 1000 y 3000 litros por día
   - **Alto**: más de 3000 litros por día

---

## 3. Análisis del problema

### 3.1 Entrada del sistema
El programa debe recibir:

- número de plumas asignadas al hogar.

### 3.2 Proceso
El programa debe:

- convertir el número de plumas en caudal total;
- transformar el caudal a consumo diario;
- transformar el consumo diario a consumo mensual;
- clasificar el nivel de consumo.

### 3.3 Salida
El programa debe mostrar:

- caudal total;
- consumo diario;
- consumo mensual;
- clasificación del consumo.

## 4. Variables y constantes sugeridas

El estudiante puede trabajar con variables como las siguientes:

- `plumas`
- `caudal_pluma`
- `caudal_total`
- `segundos_dia`
- `consumo_diario`
- `consumo_mensual`
- `nivel_consumo`

> Puede cambiar los nombres, pero deben ser claros y significativos.

## 5. Algoritmo propuesto

Antes de programar, escriba el algoritmo en lenguaje natural o seudocódigo.

### Algoritmo en lenguaje natural

1. Solicitar al usuario el número de plumas de agua.
2. Definir el valor de una pluma en litros por segundo.
3. Definir la cantidad de segundos que tiene un día.
4. Calcular el caudal total.
5. Calcular el consumo diario.
6. Calcular el consumo mensual.
7. Determinar si el consumo es bajo, medio o alto.
8. Mostrar los resultados de manera clara.

### Seudocódigo guía

```text
Inicio
    Leer número de plumas
    Definir caudal de una pluma
    Definir segundos por día

    Calcular caudal total
    Calcular consumo diario
    Calcular consumo mensual

    Si consumo diario es menor que 1000
        nivel = "Bajo"
    Si no, si consumo diario está entre 1000 y 3000
        nivel = "Medio"
    En otro caso
        nivel = "Alto"

    Mostrar caudal total
    Mostrar consumo diario
    Mostrar consumo mensual
    Mostrar nivel
Fin
```

## 6. Espacio para que el estudiante construya la solución

### Instrucciones
Complete el siguiente bloque de código a partir del algoritmo anterior.

- No copie una solución terminada sin comprenderla.
- Escriba comentarios propios.
- Use nombres de variables comprensibles.
- Verifique que los resultados tengan sentido físico.

In [1]:
# ==========================================================
# ACTIVIDAD: MODELAR LA "PLUMA DE AGUA" CON PYTHON
# ==========================================================

# ENTRADA: Solicitar al usuario el número de plumas
plumas = float(input("Ingrese el número de plumas: "))

#PROCESO: Definir constantes del problema

##Definir constantes del problema

Caudal_pluma= 0.025 # litros por segundo

segundos_dia= 24 * 60 * 60 # segundos

dias_mes= 30 # dias

## Calcular el caudal total de agua que fluye a través de todas las plumas asignadas.
caudal_total= plumas * Caudal_pluma # litros/segundo

##Calcular el consumo diario

consumo_diario= caudal_total * segundos_dia # litros/dia

##Calcular el consumo mensual

consumo_mensual= consumo_diario * dias_mes # litros/mes

## Clasificar el consumo
nivel_consumo=""

if consumo_diario < 1000:
  nivel_consumo = "Bajo"
elif 1000 <= consumo_diario <= 3000:
  nivel_consumo = "Medio"
else:
  nivel_consumo = "Alto"

# SALIDA: imprimir el resultado

## Mostrar resultados
print(f" Resultados del Consumo de Agua ")
print(f"Número de plumas ingresadas: {plumas}")
print(f"Caudal total: {caudal_total:.3f} litros por segundo")
print(f"Consumo diario: {consumo_diario:.2f} litros")
print(f"Consumo mensual: {consumo_mensual:.2f} litros")
print(f"Clasificación del consumo: {nivel_consumo}")



Ingrese el número de plumas: 8
 Resultados del Consumo de Agua 
Número de plumas ingresadas: 8.0
Caudal total: 0.200 litros por segundo
Consumo diario: 17280.00 litros
Consumo mensual: 518400.00 litros
Clasificación del consumo: Alto


## 7. Verificación conceptual

Antes de dar por terminado el programa, responda en comentarios o en una celda Markdown:

1. ¿Qué representa físicamente la variable principal del problema?
2. ¿Por qué este ejercicio conecta historia, cultura e ingeniería?
3. ¿Qué operaciones aritméticas fueron necesarias?
4. ¿Qué estructura condicional utilizó y por qué?
5. ¿Cómo interpretaría el resultado en un contexto real?

##  **Respuestas a la Verificación Conceptual**
1.   **¿Qué representa físicamente la variable principal del problema?**

La variable clave `plumas` refleja literalmente la cantidad de 'plumas de agua' que se otorgan a una vivienda, siendo esta una medida tradicional de flujo. Cada pluma corresponde a un flujo determinado en litros cada segundo.

2.   **¿Por qué este ejercicio conecta historia, cultura e ingeniería?**

Este ejercicio une la historia al emplear la expresión 'pluma de agua', que tiene su origen en costumbres de la época colonial. También se relaciona con la cultura al señalar que en el Caribe colombiano se utiliza 'pluma' para designar el grifo del agua, un legado de esa historia. Finalmente, se vincula con la ingeniería al representar un concepto físico (flujo) y sus cambios (consumo diario/mensual) mediante fórmulas y razonamiento programático, que son fundamentos de la ingeniería.

3.  **¿Qué operaciones aritméticas fueron necesarias?**

Se llevaron a cabo las siguientes operaciones matemáticas:
* Multiplicación (`*`): Para determinar el caudal total (plumas * caudal_pluma), el consumo diario (caudal_total * segundos_dia) y el consumo mensual (consumo_diario * dias_mes).
* División (`/`): Se aplica de manera implícita en la definición de `Caudal_pluma` (litros por segundo).
* Suma (`+`): No se empleó de manera directa en los cálculos fundamentales, aunque podría formar parte de operaciones más complicadas.

4. **¿Qué estructura condicional utilizó y por qué?**

Se implementó una estructura condicional `if-elif-else` para determinar el grado de uso (`nivel_consumo`). Este tipo de estructura es adecuada cuando se requiere examinar varias condiciones que no se superponen (bajo, medio, alto) y llevar a cabo un conjunto de instrucciones particular para la primera condición que sea verdadera. Proporciona una lógica sencilla y bien estructurada para la toma de decisiones fundamentadas en el valor de `consumo_diario`.

5. **¿Cómo interpretaría el resultado en un contexto real?**

En una situación real, los resultados nos ofrecen una perspectiva cuantitativa sobre el uso del agua. Si el `consumo_diario` es elevado (por ejemplo, 17,280 litros para 8 grifos), esto podría señalar un derroche de agua, una fuga, o que la cantidad de grifos es excesiva para un hogar estándar. Esta información resultaría esencial para:
* Administración de recursos: Las compañías de servicios públicos podrían utilizar esto para supervisar el uso de agua.
* Conciencia del usuario: Las familias podrían modificar sus comportamientos si notan que su consumo es 'Alto'.
* Políticas de ahorro: Esto podría influir en las decisiones relacionadas con tarifas o iniciativas de conservación.

El nivel 'Alto' en un consumo diario de 17,280 litros indica un uso extremadamente elevado que, para una vivienda, sería insostenible o podría señalar una irregularidad.

## 8. Extensión opcional

Si desea profundizar, puede ampliar el programa para que también permita:

- calcular el consumo semanal;
- comparar el consumo de varios hogares;
- estimar el costo del agua si se conoce una tarifa por litro o por metro cúbico;
- convertir litros a metros cúbicos;
- presentar los resultados en forma tabular.

> Estas extensiones no reemplazan la solución base; son una ampliación.

## 9. Documentación del código con apoyo de IA

Después de construir y probar su programa:

1. Copie su código en ChatGPT.
2. Solicite una **explicación técnica en Markdown** del código.
3. Pida además:
   - descripción de la función del programa;
   - explicación de cada sección;
   - explicación de variables, entradas, procesos y salidas;
   - observaciones sobre claridad y buenas prácticas.
4. Mejore su código con base en esa retroalimentación.
5. Actualice su repositorio en GitHub con la versión mejorada y documentada.

### Prompt sugerido para documentar el código

```text
Actúa como docente universitario experto en programación con Python.

Te voy a compartir un código en Python.
Necesito que expliques el código en formato Markdown de manera clara, técnica y ordenada.

Incluye:
1. propósito general del programa;
2. explicación de entradas, procesos y salidas;
3. descripción de cada variable importante;
4. explicación de las estructuras usadas;
5. observaciones sobre claridad, estilo y posibles mejoras;
6. una versión comentada del código, sin cambiar su lógica.

Aquí está mi código:
[PEGAR AQUÍ EL CÓDIGO]
```

## **Documentación del código con apoyo de IA**

### 1. Propósito general del programa

El programa tiene como objetivo calcular el consumo de agua generado por un número determinado de plumas de agua.
A partir de la cantidad de plumas ingresadas por el usuario, el sistema: calcula el caudal total; estima el consumo diario;
estima el consumo mensual; clasifica el nivel de consumo como: Bajo, Medio,
Alto.
El programa utiliza operaciones matemáticas básicas y estructuras condicionales para procesar la información.

### 2. Explicación de Entradas, Procesos y Salidas
**Entrada:**

El sistema pide al usuario un solo dato de entrada: la cantidad de grifos asignados, que indica la capacidad de flujo de agua en la vivienda. Se requiere un número, que puede ser con decimales.

**Proceso:**

Establecimiento de Valores Fijos: Se definen números constantes para el flujo de un grifo (Flujo_grifo), el total de segundos en un día (segundos_dia) y los días que hay en un mes (dias_mes).
Cálculo del Flujo Total: Se multiplica el total de grifos ingresados por el flujo de un grifo para calcular el flujo total en litros por segundo.
Cálculo del Uso Diario: Se convierte el flujo total (litros/segundo) en uso en litros por día, multiplicándolo por el número de segundos en un día.
Cálculo del Uso Mensual: Se obtiene el uso total en litros para el mes, multiplicando el uso diario por la cantidad de días en un mes.
Clasificación del Uso: Se analiza el uso_diario usando una estructura condicional (if-elif-else) para clasificar el consumo como "Bajo", "Medio" o "Alto".

**Salidas:**

El sistema presenta de manera clara y organizada los siguientes resultados:
La cantidad de grifos que el usuario ha ingresado.
El flujo_total en litros por segundo, con tres cifras decimales.
El uso_diario en litros, con dos cifras decimales.
El uso_mensual en litros, con dos cifras decimales.
La Clasificación del uso (Bajo, Medio, Alto).

### 3. Descripción de Cada Variable Importante

* `plumas` (float): Guarda la cantidad de plumas de agua que el usuario proporciona. Es la variable principal de entrada.
* `Caudal_pluma` (float): Valor constante que especifica el flujo de una pluma en litros por segundo (0. 025 L/s).
* `segundos_dia` (int): Constante que señala el número total de segundos en un día (24 * 60 * 60 = 86400 segundos).
* `dias_mes` (int): Constante que establece el número de días que se consideran en un mes para el cálculo mensual (30 días).
* `caudal_total` (float): Resultado de la suma del caudal de todas las plumas en litros por segundo.
* `consumo_diario` (float): Muestra la cantidad total de litros utilizados en un día, proveniente de `caudal_total`.
* `consumo_mensual` (float): Representa la cantidad total de litros usados en un mes, fundamentado en `consumo_diario`.
* `nivel_consumo` (str): Almacena la categoría del consumo de agua ("Bajo", "Medio", "Alto") basado en `consumo_diario`.

### 4. Explicación de las Estructuras Usadas
* Función `input()`: Se usa para recoger datos del usuario (`plumas`).
* Función `float()`: Transforma la entrada textual del usuario en un número decimal, lo que permite utilizar valores con parte fraccionaria para las plumas.
* Operadores Aritméticos: Se aplica la multiplicación (`*`) para llevar a cabo todos los cálculos relacionados con el flujo y el consumo.
* Estructura Condicional `if-elif-else`: Se usa para la lógica de categorización del consumo. Esta estructura permite analizar distintos intervalos para `consumo_diario` y asignar el `nivel_consumo` correspondiente. Es útil en situaciones donde existen varias condiciones que no se pueden dar al mismo tiempo.
* Función `print()` con f-strings: Se emplea para presentar los resultados de manera clara y comprensible. Los f-strings (`f". . . "`) hacen más sencillo incluir variables y dar formato a los números decimales con un número específico de cifras después del punto (por ejemplo, `:. 2f`).

### 5. Observaciones sobre Claridad, Estilo y Posibles Mejoras
**Claridad y Estilo:**
* Nombres de Variables: Denominaciones como `Caudal_pluma`, `segundos_dia`, `consumo_diario` son precisas y reveladoras, lo que facilita la lectura del código.
* Comentarios: El código tiene comentarios adecuados, delimitando claramente las partes de ENTRADA, PROCESO y SALIDA, así como describiendo las constantes y los cálculos. Esto es una práctica muy beneficiosa.
* Organización: La disposición del código sigue un orden lógico desde la entrada hasta la salida, lo que hace que sea más sencillo de entender.
* Formato de Salida: El empleo de f-strings para presentar la salida es una gran elección, ya que hace que los resultados sean claros y fáciles de interpretar, especialmente al redondear decimales.

**Posibles Mejoras:**
* Encapsulamiento en Funciones: Para que el código sea más modular y reutilizable, podrían agruparse los cálculos y la lógica de clasificación en una o varias funciones. Por ejemplo, una función `calcular_consumo(num_plumas)` que retorne un diccionario con todos los resultados.
* Validación de Entrada: En la actualidad, se asume que el usuario proporcionará siempre un número válido. Sería útil incorporar un bloque `try-except` para manejar errores que surjan si el usuario escribe texto en vez de un número, o implementar una validación que asegure que el número de plumas sea positivo.
* Constantes Globales: Aunque están bien establecidas, para proyectos más extensos, las constantes podrían consolidarse dentro de un objeto o un módulo de configuración.
* Flexibilidad de Días en el Mes: El `dias_mes` está fijado en 30. Podría ser ingresado por el usuario o extraído de la biblioteca `datetime` para mayor precisión en caso de un mes específico.
* Interfaz de Usuario: Para mejorar la interactividad, podría considerarse una interfaz gráfica simple (por ejemplo, `tkinter`, `PyQt`) o un sistema de bucle que permita calcular múltiples situaciones sin tener que reiniciar la aplicación.
* Umbrales de Clasificación: Los límites de 1000 y 3000 litros podrían establecerse como constantes al principio del código para facilitar su modificación.

### 6. Versión Comentada del Código (Sin Cambiar su Lógica)





In [ ]:

# ACTIVIDAD: MODELAR LA "PLUMA DE AGUA" CON PYTHON

# ENTRADA
# Solicitar al usuario el número de plumas de agua


plumas = float(input("Ingrese el número de plumas: "))

# PROCESO
# Definir constantes del problema


# Cantidad de litros por segundo que consume una pluma
Caudal_pluma = 0.025

# Cantidad de segundos que tiene un día
segundos_dia = 24 * 60 * 60

# Cantidad de días considerados en un mes
dias_mes = 30


# Calcular el caudal total de agua


# Multiplica el número de plumas por el caudal de cada una
caudal_total = plumas * Caudal_pluma


# Calcular el consumo diario


# Convierte litros/segundo a litros/día
consumo_diario = caudal_total * segundos_dia

# Calcular el consumo mensual


# Multiplica el consumo diario por los días del mes
consumo_mensual = consumo_diario * dias_mes

# Clasificar el nivel de consumo


# Variable donde se guardará la clasificación
nivel_consumo = ""

# Si el consumo es menor a 1000 litros
if consumo_diario < 1000:
    nivel_consumo = "Bajo"

# Si está entre 1000 y 3000 litros
elif 1000 <= consumo_diario <= 3000:
    nivel_consumo = "Medio"

# Si supera los 3000 litros
else:
    nivel_consumo = "Alto"


# SALIDA


print(f"Resultados del Consumo de Agua")
print(f"Número de plumas ingresadas: {plumas}")
print(f"Caudal total: {caudal_total:.3f} litros por segundo")
print(f"Consumo diario: {consumo_diario:.2f} litros")
print(f"Consumo mensual: {consumo_mensual:.2f} litros")
print(f"Clasificación del consumo: {nivel_consumo}")


## 10. Trabajo con GitHub

Una vez terminado el programa:

1. Cree un repositorio en GitHub.
2. Suba el notebook o el archivo `.py`.
3. Haga un primer commit.
4. Después de ampliar la documentación del código, haga un segundo commit.
5. Copie la URL del repositorio y consérvela para la entrega.

### Ejemplo de estructura del repositorio

```text
actividad-pluma-agua/
│
├── README.md
├── pluma_agua.py
└── Actividad_Pluma_de_Agua_Algoritmo_GitHub.ipynb
```

## 11. Comandos Git sugeridos

> **Nota:** estos comandos deben ejecutarse en la terminal, Git Bash o PowerShell.

### 11.1 Inicializar el repositorio local

```bash
git init
git add .
git commit -m "Versión inicial de la actividad pluma de agua"
```

### 11.2 Conectar con GitHub

```bash
git branch -M main
git remote add origin URL_DEL_REPOSITORIO
git push -u origin main
```

### 11.3 Subir una segunda versión con mejor documentación

```bash
git add .
git commit -m "Se mejora documentación y comentarios del código"
git push
```

## 12. Clonar el repositorio en el disco local C:

Una vez que el repositorio esté en GitHub, copie su URL y ejecute:

```bash
cd /c
git clone URL_DEL_REPOSITORIO
```

Si está usando PowerShell en Windows, también puede hacerlo así:

```powershell
cd C:\
git clone URL_DEL_REPOSITORIO
```

### Ejemplo esperado

Si su repositorio se llama:

```text
https://github.com/usuario/actividad-pluma-agua.git
```

entonces el comando sería:

```bash
git clone https://github.com/usuario/actividad-pluma-agua.git
```

Con esto se creará una carpeta del proyecto dentro de `C:\`.

## 13. Evidencia de entrega sugerida

El estudiante debe entregar:

- el enlace del repositorio en GitHub;
- el notebook o archivo `.py` funcional;
- una versión documentada del código;
- evidencia de que el repositorio fue clonado en el equipo local.

### Evidencia mínima recomendada
- captura del repositorio en GitHub;
- captura de los commits;
- captura del comando `git clone` ejecutado correctamente.

## 14. Reflexión final

Este ejercicio no trata solamente de programar.  
También muestra cómo una palabra cotidiana del Caribe colombiano puede llevarnos a:

- la historia de la hidráulica;
- la representación computacional de una magnitud física;
- el diseño de algoritmos;
- el uso responsable de herramientas como GitHub y ChatGPT en el trabajo académico.

> La programación también puede nacer de la memoria cultural y del contexto local.

## 15. Espacio para notas del estudiante

Escriba aquí una breve reflexión personal:

- ¿Qué aprendió?
- ¿Qué parte le resultó más difícil?
- ¿Qué mejoró cuando documentó el código?
- ¿Qué importancia tiene publicar y versionar su trabajo?